permission_prompt/security_privacy_self_review.md

```diff
@@ -0,0 +1,108 @@
+# Security and Privacy Self-Review: Private Network Access Permission Prompt
+
+Based on the W3C TAG's
+[questionnaire](https://www.w3.org/TR/security-privacy-questionnaire/).
+
+## 1. What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+
+Documents might be able to infer whether they were loaded from `public`,
+`private` or `local` IP address by observing whether they can successfully load
+a well-known subresource from the private network or localhost. This is largely
+unavoidable by design: this specification aims to alter user agent behavior
+differentially based on the IP address which served the main resource.
+
+This can also arise when a client loaded a resource over a non-public proxy,
+such as:
+
+* an ssh tunnel
+* a network inspection tool such as [Fiddler](https://telerik.com/fiddler)
+* a corporate proxy
+
+Timing attacks might also be possible, though no concrete scenario has yet been
+laid out.
+
+See [#41](https://github.com/WICG/private-network-access/issues/41) for a discussion of these points.
+
+## 2. Do features in your specification expose the minimum amount of information necessary to enable their intended uses? 
+
+Yes, apart from the above.
+
+## 3. How do the features in your specification deal with personal information, personally-identifiable information (PII), or information derived from them?
+
+It does not.
+
+## 4. How do the features in your specification deal with sensitive information?
+
+It does not.
+
+## 5. Do the features in your specification introduce new state for an origin that persists across browsing sessions?
+
+Yes, the permission status which allows a certain public webpage to access a certain private/local service.
+
+## 6. Do the features in your specification expose information about the underlying platform to origins?
+
+See question 1.
+
+## 7. Does this specification allow an origin to send data to the underlying platform?
+
+No.
+
+## 8. Do features in this specification enable access to device sensors?
+
+No.
+
+## 9. Do features in this specification enable new script execution/loading mechanisms?
+
+No.
+
+## 10. Do features in this specification allow an origin to access other devices?
+
+No, instead it restricts the ability of origins to access local network devices.
+
+## 11. Do features in this specification allow an origin some measure of control over a user agent’s native UI?
+
+No.
+
+## 12. What temporary identifiers do the features in this specification create or expose to the web?
+
+None.
+
+## 13. How does this specification distinguish between behavior in first-party and third-party contexts?
```

Comment:
This should probably also discuss how the permission prompt can only be presented by first parties?

Reply:
Good idea! We should restrict usage of this API and permission to first parties (i.e. add this to the explainer) and mention it here.

Reply:
Good point here. I would suggest that @annevk the "first parties" you are talking here is about the permission policies. My idea is that we can allow both top-level document and same-origin frames be able to raise up the permission by default. In the future, we might be able to allow top-level document delegating the permission to third-party frames. WDYT?

Reply:
I think we landed on top-level documents only at first, suggested an edit below to clarify that.

Reply:
Yes, what @letitz said is correct. We did an offline discuss and decide to land on top-level documents only first. However, we are still open for options here.